### ISSUE TITLE
Set up API connection from Medallia to BigQuery

### ISSUE THREAD

**Issue by joanneesteban** [department-of-veterans-affairs/va.gov-team#11512]
## Issue Description
How might we programmatically connect future Medallia data to BigQuery?

---
## Tasks
- [x] Get Keifer acccess to Medallia
- [x] Get Keifer access to BigQuery
- [ ] Determine the table structure for Medallia data
- [ ] Develop programmatic approach for automating Medallia data
- [x] Connect API
- [ ] Document approaches

## Acceptance Criteria
- [ ] VSP understands how to automate Medallia data in BigQuery

**Comment by kfrz** [comment]
### Open Questions

- Potentially need a auth key?
- Can we do read-only from API?
- Blocked by lack of docs access
- Security concerns w/r/t network transit
- Example of twitter https://medium.com/@chitre.ajay/streaming-data-directly-into-bigquery-866d0d1d36f0  

### Requirements for integration to BigQuery

- auth token/param from medallia
- form schema & id
- schema for response elements

**Comment by kfrz** [comment]
Notes in above comment are updated to reflect meeting today, 07/28/20. We will meet again on 07/29/20 to determine details around export process and start testing connectivity.

**Comment by kfrz** [comment]
Medallia does not yet have GCS exporter

They are not quite there -> it's on the roadmap, GovCloud gets things last because we're on a different version. 

HAMC 

Setup an AWS account, Instead of S3

We setup a new key for the GCS bucket

Sarkis helped us to create and configure an exporter using dummy data on the SBX2 server

**Comment by kfrz** [comment]
**07/31/20**

- Created a `TEST_MEDALLIA_RESPONSE_TABLE` in [Google Cloud Storage](https://console.cloud.google.com/bigquery?project=vsp-analytics-and-insights&p=vsp-analytics-and-insights&d=test_medallia_vsp&t=TEST_MEDALLIA_RESPONSE_TABLE&page=table)
- Imported the dummy data from Medallia to test schema, this worked and BigQuery was able to detect schema automatically
- Attempted to discover how best to automatically ingest Cloud Storage files to Big Query table
- Discovered Google Dataflow pipelines, created [test job](https://console.cloud.google.com/dataflow/jobs/us-central1/2020-07-31_12_55_22-12419481703757038532?project=vsp-analytics-and-insights)
- This failed because (we think) it was using a TextIO template, which references specific text attributes.


**Potential Pathway to Pub/Sub Success**

Meeting on 08/03/20 and will attempt to wire up Pub/Sub Dataflow

- create pub/sub topic for medallia-bucket-upload changes [see docs](https://cloud.google.com/storage/docs/reporting-changes)
- add rule to bucket for upload notification
- create message on upload
- create a subscription topic for upload
- create job using PubSubToBigQuery [template](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/master/src/main/java/com/google/cloud/teleport/templates/PubSubToBigQuery.java)

**Comment by alexpappasoddball** [comment]
Removing from BE Tools board.  If any more help is needed we can add it back or create a new ticket with specific tasks.

**Comment by jonwehausen** [comment]
Historical ticket that can now be closed as we're actively working on this as an initiative
